clusterloader2/testing/experimental/storage/pod-startup/config.yaml

```diff
@@ -7,12 +7,22 @@
   {{$NODES_PER_NAMESPACE := DefaultParam .NODES_PER_NAMESPACE 100}}
 
 # Test Variales
-  {{$PODS_PER_NODE := .PODS_PER_NODE}}
+  {{$PODS_PER_NODE := DefaultParam .PODS_PER_NODE 1}}
   {{$DEPLOYMENT_TEMPLATE_PATH := .DEPLOYMENT_TEMPLATE_PATH }}
-  {{$VOLUMES_PER_POD := .VOLUMES_PER_POD}}
+  {{$VOLUMES_PER_POD := DefaultParam .VOLUMES_PER_POD 1}}
   {{$VOLUME_TEMPLATE_PATH := .VOLUME_TEMPLATE_PATH}}
+  # Set this to false if apiserver, kube-controller-manager and kube-scheduler
+  # do not export metrics data via the insecure http ports.
```

Comment:
I ran against a cluster where "kubeadm" locked down access to metrics via http and then gathering those metrics in clusterload2 failed because it couldn't connect to the ports.

The code which pulls metrics uses the insecure ports: https://github.com/kubernetes/kubernetes/blob/e799c852fb337be414fa2dae982fed69a67806fe/test/e2e/framework/metrics/metrics_grabber.go#L35-L44

Reply:
kube-apiserver indeed works without the insecure port.

But a missing insecure port for kube-controller-manager causes the test to fail:
```
...
I1013 17:00:22.539742  220515 resource_gather_worker.go:69] Get container kube-controller-manager-pmem-csi-pmem-govm-master/kube-controller-manager usage on node pmem-csi-pmem-govm-master. CPUUsageInCores: 0.011748654, MemoryUsageInBytes: 0, MemoryWorkingSetInBytes: 44888064
I1013 17:00:22.539758  220515 resource_gather_worker.go:69] Get container kube-proxy-7474b/kube-proxy usage on node pmem-csi-pmem-govm-master. CPUUsageInCores: 0.000631058, MemoryUsageInBytes: 0, MemoryWorkingSetInBytes: 13516800
E1013 17:00:30.651327  220515 metrics_for_e2e.go:79] MetricsForE2E: metricsGrabber failed to grab some of the metrics: Errors while grabbing metrics: [error waiting for controller manager pod to expose metrics: timed out waiting for the condition; an error on the server ("unknown") has prevented the request from succeeding (get pods kube-controller-manager-pmem-csi-pmem-govm-master:10252)]
...
```

Reply:
OK - looked into the code and I know what is happening (I was wondering why this works for GKE where we don't even expose access to pods)

And apparently - we're grabbing those metrics via listing pods and if we don't find pods, we simply ignore them:
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/metrics/metrics_grabber.go#L98

This code will have to be rewritten anyway (it e.g. doesn't support HA setups).

But in the meantime, the correct way to solve this would be to:
- introducing new features similar to SupportGrabMetricsFromKubelets in provider package:
https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/pkg/provider/provider.go#L52
- using this in the measurement similarly as we do for kubelets:
https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/pkg/measurement/common/metrics_for_e2e.go#L63

That should address your problem and allow you to run all the other measurements that are actually useful.